File: FullFlowExamples/python/PaymentForm_PayWithCCToken.py
import splitit
from splitit.rest import ApiException

username, password, api_key = (
    'username',
    'password',
    'api_key'
)

api_client = splitit.ApiClient(splitit.Configuration(username, password, api_key, sandbox=True))


def get_cc_token(old_installment_plan_no):
    """For a returning shopper, use an existing plan that is related to that shopper"""
    resp = api_client.InstallmentPlanApi.installment_plan_get(
        splitit.GetInstallmentsPlanSearchCriteriaRequest(
            splitit.InstallmentPlanQueryCriteria(
                installment_plan_number=old_installment_plan_no,
            )
        )
    )
    return resp.plans_list[0].active_card.token


def initiate_new_plan():
    """Set a new installment plan for the returning shopper"""
    # TODO: Fill-in the data
    resp = api_client.InstallmentPlanApi.installment_plan_initiate(
        splitit.InitiateInstallmentPlanRequest(
            plan_data=splitit.PlanData(
                auto_capture=True,
                amount=splitit.MoneyWithCurrencyCode(300, "USD"),
                number_of_installments=3,
                ref_order_number="abc123",
            ),
            payment_wizard_data=splitit.PaymentWizardData(
                requested_number_of_installments="2,3,4,5,6",
            ),
            billing_address=splitit.AddressData2(
                address_line="260 Madison Avenue.",
                city="New York",
                state="NY",
                country="USA",
                zip="10016",
            ),
            consumer_data=splitit.ConsumerData(
                full_name="John Smith",
                email="JohnS@splitit.com",
                phone_number="1-415-775-4848",
                culture_name="en-us",
            ),

        )
    )
    return resp


def pay_with_token(new_plan_number, cc_token):
    """Create the installment plan using Credit Card Token"""
    # TODO: Customize the data
    resp = api_client.InstallmentPlanApi.installment_plan_create(
        splitit.CreateInstallmentPlanRequest(
            installment_plan_number=new_plan_number,
            plan_data=splitit.PlanData(
                amount=splitit.MoneyWithCurrencyCode(300, "USD"),
            ),
            payment_token=splitit.PaymentToken(
                token=cc_token,
                type="SplititStoredCard",
            )
        )
    )
    print("Successfully payed for the plan")
    return resp

File: FullFlowExamples/java/PaymentFormPayWithCCToken.java
import com.splitit.*;
import com.splitit.sdk.model.*;
import java.math.BigDecimal;
import java.math.RoundingMode;
import org.threeten.bp.OffsetDateTime;


public class PaymentFormPayWithCCToken {
    private ApiClient apiClient;
    public PaymentFormPayWithCCToken(String username, String password, String apiKey){
        this.apiClient = new ApiClient()
            .sandbox(true)
            .username(username)
            .password(password)
            .apiKey(apiKey);
    }

    // For a returning shopper, use an existing plan that is related to that shopper
    public String getCCToken(String oldInstallmentPlanNumber) throws ApiException{
        var resp = apiClient.getInstallmentPlanApi().installmentPlanGet(
            new GetInstallmentsPlanSearchCriteriaRequest()
            .queryCriteria(
                new InstallmentPlanQueryCriteria()
                .installmentPlanNumber(oldInstallmentPlanNumber)
            )
        );
        return resp.getPlansList().get(0).getActiveCard().getToken();
    }


    // Set a new installment plan for the returning shopper
    public InitiateInstallmentsPlanResponse initiateNewPlan() throws ApiException{
        // TODO: Fill-in the data
        InitiateInstallmentsPlanResponse resp = apiClient.getInstallmentPlanApi().installmentPlanInitiate(
            new InitiateInstallmentPlanRequest()
            .planData(
                new PlanData()
                .autoCapture(true)
                .amount(
                    new MoneyWithCurrencyCode()
                    .value(new BigDecimal("300"))
                    .currencyCode("USD")
                )
                .numberOfInstallments(3)
                .refOrderNumber("abc123")
            )
            .paymentWizardData(
                new PaymentWizardData()
                .requestedNumberOfInstallments("2,3,4,5,6")
            )
            // Optional data to pre-fill the form
            .billingAddress(
                new AddressData()
                .addressLine("260 Madison Avenue.")
                .city("New York")
                .state("NY")
                .country("USA")
                .zip("10016")
            )
            .consumerData(
                new ConsumerData()
                .fullName("John Smith")
                .email("JohnS@splitit.com")
                .phoneNumber("1-415-775-4848")
                .cultureName("en-us")
            )
        );
        return resp;
    }

    // Create the installment plan using Credit Card Token
    public void payWithToken(String installmentPlanNumber, String ccToken) throws ApiException{
        var resp = apiClient.getInstallmentPlanApi().installmentPlanCreate(
            new CreateInstallmentPlanRequest()
            .installmentPlanNumber(installmentPlanNumber)
            .planData(
                new PlanData()
                .amount(
                    new MoneyWithCurrencyCode()
                    .value(new BigDecimal("300"))
                    .currencyCode("USD")
                )
            )
            .paymentToken(
                new PaymentToken()
                .token(ccToken)
                .type("SplititStoredCard")
            )
        );

        System.out.println("Successfully payed for the plan");
    }
}
